osidb/models.py

```diff
@@ -2919,6 +2933,76 @@ def validate(self, *args, **kwargs):
         super().clean_fields(*args, **kwargs)
 
 
+class PackageManager(ACLMixinManager, TrackingMixinManager):
+    @staticmethod
+    def create_package(flaw, package, **extra_fields):
+        """
+        Return a new Package, or update an existing Package without saving.
+        Unlike other similar manager methods in this file, new Package
+        instance is saved into database to allow relationships with PackageVer.
+        """
+        try:
+            package = Package.objects.get(flaw=flaw, package=package)
+            for attr, value in extra_fields.items():
+                setattr(package, attr, value)
+            return package
+
+        except ObjectDoesNotExist:
+            package = Package(flaw=flaw, package=package, **extra_fields)
+            package.save()
+            return package
+
+
+class Package(ACLMixin, BugzillaSyncMixin, TrackingMixin):
+    """
+    Model representing a package with connected versions.
+
+    The model's structure allows future extensibility with features from the CVE 5.0 schema.
+    Currently, it tracks only versions corresponding to bugzilla fixed_in field, which correspond to
+    CVE 5.0's `"status": "unaffected"`.
+    References:
+    - https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/docs/versions.md
+    - https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json
+    """
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    flaw = models.ForeignKey(
+        Flaw,
+        related_name="package_versions",
+        on_delete=models.CASCADE,
+    )
+
+    package = models.CharField(max_length=2058)
+
+    objects = PackageManager()
+
+    def bzsync(self, *args, bz_api_key, **kwargs):
+        """
+        Bugzilla sync of the Package instance and linked PackageVer instances.
+        """
+        self.save()
+        # needs to be synced through flaw
+        self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
+
+
+class PackageVer(models.Model):
```

Comment:
What about `PackageVersion`?

Reply:
That model name is already used. Renaming models is complicated. Additional context [in the first PR comment](https://github.com/RedHatProductSecurity/osidb/pull/307#issue-1855463280) and in [this comment](https://github.com/RedHatProductSecurity/osidb/pull/307#discussion_r1319485140).

Reply:
Okay, but that's so close that I didn't even notice. That would be confusing IMO, even though the old models should be deleted in a month or so.